nRF9160Master/src/Temp_MCP/TempAdcHandler.c

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file    Temp_Adchandler.c
+ * @author : Devendu
+ * @brief   File to calculate the temperature using MCP9700
+ * @date    09-04-2023
+ */
+/***********************************INCLUDES**************************/
+#include <nrfx_example.h>
+#include <saadc_examples_common.h>
+#include <nrfx_saadc.h>
+#include <zephyr/kernel.h>
+#include "TempAdcHandler.h"
+
+
+/************************************MACROS***************************/
+#define NRFX_LOG_MODULE                 EXAMPLE
+#define NRFX_EXAMPLE_CONFIG_LOG_ENABLED 1
+#define NRFX_EXAMPLE_CONFIG_LOG_LEVEL   3
+#include <nrfx_log.h>
+
+/** @brief Symbol specifying analog input to be observed by SAADC channel 0. */
+#define CH0_AIN ANALOG_INPUT_TO_SAADC_AIN(ANALOG_INPUT_A0)
+
+// Temperature coefficient in mV/°C
+#define TC_MV_PER_C 10.0
+
+// Output voltage at 0°C in mV
+#define V0_C_MV 440
+
+
+/************************************GLOBALS**************************/
+
+/** @brief Array specifying GPIO pins used to test the functionality of SAADC. */
+static uint8_t m_out_pins[1] = {LOOPBACK_PIN_1B, LOOPBACK_PIN_2B, LOOPBACK_PIN_3B};
+
+/** @brief Samples buffer defined with the size of @ref CHANNEL_COUNT symbol to store values from each channel ( @ref m_multiple_channels). */
+static nrf_saadc_value_t samples_buffer[1];
+
+/** @brief SAADC channel configuration structure for single channel use. */
+static const nrfx_saadc_channel_t m_single_channel = NRFX_SAADC_DEFAULT_CHANNEL_SE(CH0_AIN, 0);
+
+
+
+/**********************************FUNCTION DEFINITIONS****************/
+
+/**
+ * @brief Converts raw ADC value to temperature using the MCP9700 formula.
+ *
+ * This function takes a raw ADC value as input and converts it to temperature using the MCP9700 formula. 
+   It assumes a 10-bit resolution and a reference voltage of 3.3V.
+ *
+ * @param adc_value The raw ADC value to be converted to temperature.
+ * @return float - Temperature calculated from the ADC value, in degrees Celsius.
+ */
+
+static float adc_to_temperature(uint16_t adc_value) {
+    // Convert ADC value to voltage (assuming 10-bit resolution and Vref = 3.3V)
+    float voltage = (float)adc_value * 3.3 / 1024.0;
```

Comment:
Code should be readable and easy to maintain. Please declare and define the variables at the start of function.

Reply:
okay